meeshkan/serve/mock/data/storage.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import uuid
+
+from http_types import Request
+from openapi_typed import OpenAPIObject
+
+logger = logging.getLogger(__name__)
+
+
+class Storage:
+    def __init__(self):
+        self._default = dict()
+        self._entities = dict()
+
+    @property
+    def default(self):
+        return self._default
+
+    def add_entity(self, name):
+        self._entities[name] = dict()
+
+    def clear(self):
+        self._default.clear()
+        for entity in self._entities.values():
+            entity.clear()
+        logger.debug("Storage cleared")
+
+    def __getattr__(self, x):
+        if x in self.__dict__:
+            return getattr(self, x)
+        return self._entities[x]
+
+    def __getitem__(self, x):
+        return self._entities[x]
+
+    def __contains__(self, x):
+        return x in self._entities
+
+
+
+
+class StorageManager:
```

Comment:
Maybe it'd be a good time now to rename this class to something like `StorageStore`. That way, the understanding going forward is that it is a collection of storages.

Reply:
We have CallbackManager for a collection of callbacks, RestMiddlewareManager for a collections of middleware and this one. We may rename them all but I think "manager" is fine here, because it's assumed that these classes will manage lifecycles if necessary.

Reply:
I think currently they don't manage lifecycles though? I agree a `Store` or similar would make it clearer what they do but it doesn't need to be part of this PR.

Reply:
Renamed to MockDataStore